### PR TITLE
[IOTDB-6069] Support SessionPool in client-py

### PR DESCRIPTION
**Pull request by ycycse**
This PR implement a simplify  session pool for python client.
```python
"""
 PoolConfig(host: str, ip: str, user_name: str, password: str, node_urls: list = None,
                 fetch_size: int = DEFAULT_FETCH_SIZE, time_zone: str = DEFAULT_TIME_ZONE,
                 max_retry: int = DEFAULT_MAX_RETRY)
"""
pool_config = PoolConfig(db.get_ip_address(), db.get_port(), [], "root", "root", 1024, "Asia/Shanghai", 3)

"""
create_session_pool(pool_config: PoolConfig, max_pool_size: int, wait_timeout_in_ms: int)
"""
session_pool = create_session_pool(pool_config, max_pool_size, 3000)

# Try to get session from pool, if there isn't enough connection, a error about timeout will be raised
try:
    session = session_pool.get_session()
except TimeoutError as e:
    raise e
session.open(False)
session_pool.put_back(session)
```

### CODE REVIEW

**Review by HTHou — APPROVED** [review]

File: iotdb-client/client-py/iotdb/SessionPool.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                self.__pool_size += 1
+                should_create = True
+                self.__lock.release()
+                break
+            else:
+                if time.time() - start > self.__wait_timeout_in_ms:
+                    self.__lock.release()
+                    raise TimeoutError("Wait to get session timeout in SessionPool, current pool size: {0}"
+                                       .format(self.__max_pool_size))
+                time.sleep(1)
```
Comment (JackieTien97) [inline iotdb-client/client-py/iotdb/SessionPool.py:90]: sleep for 1 second? Make sure default time unit in python sleep is second.

File: iotdb-client/client-py/iotdb/SessionPool.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+
+class SessionPool(object):
+
+    def __init__(self, pool_config: PoolConfig, max_pool_size: int, wait_timeout_in_ms: int):
+        self.__pool_config = pool_config
+        self.__max_pool_size = max_pool_size
+        self.__wait_timeout_in_ms = wait_timeout_in_ms / 1000
+        self.__pool_size = 0
+        self.__queue = Queue(max_pool_size)
```
Comment (JackieTien97) [inline iotdb-client/client-py/iotdb/SessionPool.py:55]: Is this `Queue` a thread-safe class?

File: iotdb-client/client-py/iotdb/SessionPool.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            self.__queue.put(session)
+        else:
+            self.__lock.acquire()
+            self.__pool_size -= 1
+            self.__lock.release()
+
+    def close(self):
+        while not self.__queue.empty():
+            session = self.__queue.get(block=False)
+            session.close()
```
Comment (JackieTien97) [inline iotdb-client/client-py/iotdb/SessionPool.py:116]: don't need to update `__pool_size`?

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: iotdb-client/client-py/iotdb/SessionPool.py (hunk 1 above)
Comment (ycycse) [reply iotdb-client/client-py/iotdb/SessionPool.py:90]: Yes, the unit of time.sleep() is second. 
https://docs.python.org/3/library/time.html?highlight=sleep#time.sleep

File: iotdb-client/client-py/iotdb/SessionPool.py (hunk 2 above)
Comment (ycycse) [reply iotdb-client/client-py/iotdb/SessionPool.py:55]: It's thread-safe, but I found it not process-safe, so it should be changed.

File: iotdb-client/client-py/iotdb/SessionPool.py (hunk 3 above)
Comment (ycycse) [reply iotdb-client/client-py/iotdb/SessionPool.py:116]: This closes the sessionPool, so I think it's not necessary to update pool_size.

Comment (JackieTien97) [reply iotdb-client/client-py/iotdb/SessionPool.py:116]: But there is no guarantee that users won't use the closed SessionPool. What if they use it after closing it?

Comment (ycycse) [reply iotdb-client/client-py/iotdb/SessionPool.py:126]: I see. I have added a `closed` status for SessionPool

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**